### PR TITLE
Download Method Refactor

### PR DESCRIPTION
**Pull request by Saurav-D**
# Peer Review Information

 - Moved `_download_file` method from `datasetinsights.data.simulations.download` to `datasetinsights.data.download`
 - Made `_download_file method` public (now `download_file`)
 - Moved `DownloadError` from  `datasetinsights.data.simulations.exceptions` to `datasetinsights.data.exceptions`

# Pull Request Check List

- [x] Updated **tests** for changed code.

### CODE REVIEW

File: datasetinsights/data/download.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                f"be completed."
+            )
+
+            raise DownloadError(err_msg)
+        else:
+            dest_path.parent.mkdir(parents=True, exist_ok=True)
+            with open(dest_path, "wb") as f:
+                f.write(response.content)
+
+    return dest_path
```
Comment (BlairLee) [inline datasetinsights/data/download.py:66]: Can we rename this file name, like `http_download.py`? It's wired to have the same name with `datasetinsights/data/simulation/download.py`.

Comment (Saurav-D) [reply datasetinsights/data/download.py:66]: yes, we can do that. But this is contains generic download function on the other hand, `datasetinsights/data/simulation/download.py` contains download functions related to USIM so we can also change that to `usim_download.py`

Comment (BlairLee) [reply datasetinsights/data/download.py:66]: Ok. Sounds good!

Comment (Saurav-D) [reply datasetinsights/data/download.py:66]: I talked with YC about this, and we think it's fine to have `download.py` in both places.

Comment (BlairLee) [reply datasetinsights/data/download.py:66]: Okay. I'm okay with that.

**Review by BlairLee — APPROVED** [review]

**Pull request merged**